optima/results.py

```diff
@@ -1028,50 +1078,86 @@ def exporttoexcel(filename=None, outdict=None):
+        formats['number'] = workbook.add_format({'bg_color': colors['fadedstrawberry'], 'num_format': 0x04})
+        formats['percent'] = workbook.add_format({'bg_color': colors['fadedstrawberry'], 'num_format': 0x09})
         formats['header'] = workbook.add_format({'bg_color': colors['gentlegreen'], 'color': colors['white'], 'bold': True})
+        formats['increase'] = workbook.add_format({'bg_color': colors['increasegreen'], 'num_format': 0x09})
+        formats['decrease'] = workbook.add_format({'bg_color': colors['decreasered'], 'num_format': 0x09})
         
         # Convert from a string to a 2D array
         outlist = []
         for line in outstr.split('\n'):
             outlist.append([])
-            for cell in line.split(','):
-                if cell=='tot': cell = 'Total' # Hack to replace internal key with something more user-friendly
-                outlist[-1].append(str(cell)) # If unicode, doesn't work
+            if '"' in line:
+                for cell in line.split('","'):  # Hack to not separate program names with commas in them
+                    outlist[-1].append(str(cell))  # If unicode, doesn't work
+            else:
+                for cell in line.split(','):
+                    if cell == 'tot': cell = 'Total'  # Hack to replace internal key with something more user-friendly
+                    outlist[-1].append(str(cell))  # If unicode, doesn't work
         
         # Iterate over the data and write it out row by row.
-        row, col = 0, 0
         maxcol = 0
         for row in range(len(outlist)):
             for col in range(len(outlist[row])):
-                maxcol = max(col,maxcol)
+                maxcol = max(col, maxcol)
                 thistxt = outlist[row][col]
                 thisformat = 'plain'
-                emptycell = not(thistxt)
+                emptycell = not thistxt
+                percentcell = False
+                percentincrease = False
+                percentdecrease = False
                 try: 
                     thistxt = float(thistxt)
                     numbercell = True
-                except:
+                except ValueError:
+                    if 'perc' in thistxt:
+                        try:
+                            thistxt = thistxt.replace("perc", "")
+                            thistxt = float(thistxt)
+                            percentcell = True
+                        except ValueError:
+                            try:
+                                thistxt = thistxt.replace("conditional", "")
+                                thistxt = float(thistxt)
+                                percentcell = True
+                                if thistxt > 0:
+                                    percentincrease = True
+                                elif thistxt < 0:
+                                    percentdecrease = True
+                            finally:
+                                pass
```

Comment:
Rather than do this with a series of try-catches, I'd just check directly, and I'd probably also make the labels more unique (since e.g. a program name could easily have 'perc' in it):
```
percstr = '!PERC' # define above
condstr = '!COND' # ditto
percentcell = percstr in thistxt # will be true if found, false otherwise, and don't need to define above
conditionalcell = condstr in thistxt
thistxt = thistxt.replace(percstr, '').replace(condstr, '')
try:
  thistxt = float(thistxt)
  numbercell = True
except:
  numbercell = False
```

Reply:
Agreed, will update!